File: backend/src/main/java/com/tarantulapp/entity/PartnerProgramTier.java
package com.tarantulapp.entity;

public enum PartnerProgramTier {
    /** Curated external business with authorized catalog sync and external checkout handoff. */
    OFFICIAL_PARTNER,
    /** Limited premium official partner slot with stronger placement and co-marketing. */
    FOUNDING_PARTNER,
    /** @deprecated Use {@link #FOUNDING_PARTNER}. Kept for rows created before the ecosystem refactor. */
    @Deprecated
    STRATEGIC_FOUNDER,
    /** @deprecated Use {@link #OFFICIAL_PARTNER}. Kept for rows created before the ecosystem refactor. */
    @Deprecated
    STRATEGIC_PARTNER;

    public boolean isFoundingPartner() {
        return this == FOUNDING_PARTNER || this == STRATEGIC_FOUNDER;
    }

    public boolean isOfficialPartner() {
        return this == OFFICIAL_PARTNER || this == STRATEGIC_PARTNER || isFoundingPartner();
    }
}

File: frontend/src/utils/partnerProgramTier.js
/** Official/founding partner tier helpers (legacy STRATEGIC_* still read during DB migration). */

export function isFoundingPartnerTier(entity) {
  if (!entity) return false
  if (entity.isFoundingPartner) return true
  const tier = entity.partnerProgramTier
  return tier === 'FOUNDING_PARTNER' || tier === 'STRATEGIC_FOUNDER'
}

export function isOfficialPartnerTier(entity) {
  if (!entity) return false
  const tier = entity.partnerProgramTier
  if (tier === 'OFFICIAL_PARTNER' || tier === 'STRATEGIC_PARTNER') return true
  return isFoundingPartnerTier(entity)
}
